src/main/java/org/apache/sling/testing/mock/jcr/MockNodeType.java

```diff
@@ -54,6 +54,12 @@ public boolean hasOrderableChildNodes() {
         // support only well-known built-in node type
         return StringUtils.equals(getName(), JcrConstants.NT_UNSTRUCTURED);
     }
+    
+        @Override
+    public boolean isMixin() {
+        // if it has at least 1 supertype -> isMixin
+       return (getDeclaredSupertypes().length > 0);
```

Comment:
this method is not covered by unit tests - and it will fail for sure, as it references getDeclaredSupertypes which throws an UnsupportedOperationException.

Reply:
What is the desired approach here? 
If method `isMixin` is not critical I can undo the changes or hardcode to always return true.
Implementing `getDeclaredSupertypes` implies implementing at least 2 other methods `getDefinition` and `getDeclaredSupertypeNames` and maybe more

Reply:
Also, I've refactored the code using streams

Reply:
from my pov it's fine to move isMixin back to unsupported methods and remove the implementation. it would require a bit more rework to make this working.

Reply:
reverted changes